Packet++/src/SomeIpLayer.cpp

```diff
@@ -0,0 +1,349 @@
+#define LOG_MODULE PacketLogModuleSomeIpLayer
+
+#include "SomeIpLayer.h"
+#include "EndianPortable.h"
+#include "Packet.h"
+#include "PayloadLayer.h"
+#include <algorithm>
+#include <sstream>
+#include <string.h>
+
+namespace pcpp
+{
+std::unordered_set<uint16_t> SomeIpLayer::mSomeIpPorts{};
+
+void SomeIpLayer::optionallyCopyPayloadData(const uint8_t *const data, size_t dataLen, bool dontCopy)

```

Comment:
Assuming `SomeIpLayer` c'tor doesn't include the payload, this method can probably be removed and the simple initiation of  `m_Data`, `m_DataLen` and `m_Protocol` can be done in the c'tor itself